src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/Extensions/StringExtensions.cs

```diff
@@ -105,6 +105,30 @@ public static string RemoveBlankLines(this string value)
                 .Trim();
         }
 
+        /// <summary>
+        /// Removes the duplicate string from parameter name to work around the rosyln issue.
+        /// https://github.com/dotnet/roslyn/issues/26292.
+        /// e.g if the value is service-$Catalog-Id-1-$Catalog-Id-1, this will return
+        /// service-$Catalog-Id
+        /// </summary>
+        /// <param name="value">The param name to update.</param>
+        /// <returns>The updated value.</returns>
+        public static string RemoveDuplicateString(this string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return value;
+            }
+
+            // Find the first special character among -,$,@,_ in the string.
+            string firstOccuredSpecialCharacter = Regex.Match(value, "[-_$@]").Value;
+
+            return string.IsNullOrWhiteSpace(firstOccuredSpecialCharacter)
+                ? value
+                : string.Join(firstOccuredSpecialCharacter,
```

Comment:
Would using distinct cause a problem in this case?

variable name is a-b-b

Based on my understanding, Roslyn would give a-b-b-b-b

Using distinct will reduce this to a-b 

Can we simply remove the second half of the string after we detect the special character, if that is how the duplication works?

Reply:
yup distinct would cause what you mentioned, but then if we remove second half and param name is like $skip$test it will return $skip 

---
In reply to: [253637331](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/198#discussion_r253637331) [](ancestors = 253637331)

Reply:
also i feel less chances of have a param called a-b-b

---
In reply to: [253639007](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/198#discussion_r253639007) [](ancestors = 253639007,253637331)

Reply:
We can remove the second half only when the first half and second half is the same.

$skip$test remains $skip$test because "$skip" != "$test"

a-b-b-b-b gets truncated to a-b-b because "-b-b" == "-b-b"

Reply:
k 

---
In reply to: [253640112](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/198#discussion_r253640112) [](ancestors = 253640112)